code/modules/power/cable.dm

```diff
@@ -295,35 +302,39 @@ GLOBAL_LIST_INIT(wire_node_generating_types, typecacheof(list(/obj/structure/gri
 // Powernets handling helpers
 //////////////////////////////////////////////
 
-/obj/structure/cable/proc/get_cable_connections(powernetless_only)
+/obj/structure/cable/proc/get_cable_connections(link_dirs = linked_dirs, check_layer = cable_layer)
 	. = list()
 	var/turf/T = get_turf(src)
-	if(locate(/obj/structure/cable_bridge) in T)
-		for(var/obj/structure/cable/C in T)
-			if(C != src)
-				. += C
 	for(var/check_dir in GLOB.cardinals)
-		if(linked_dirs & check_dir)
+		if(link_dirs & check_dir)
 			T = get_step(src, check_dir)
 			for(var/obj/structure/cable/C in T)
-				if(cable_layer & C.cable_layer)
+				if(check_layer & C.cable_layer)
 					. += C
 
+/obj/structure/cable/proc/get_all_cable_connections(powernetless_only)
+	. = list()
+	var/turf/T
+	for(var/check_dir in GLOB.cardinals)
+		T = get_step(src, check_dir)
+		for(var/obj/structure/cable/C in T.contents - src)

```

Comment:
Have you profiled this, compared to `for(var/obj/structure/cable/C in T) if(C == src) continue`?

Try using parentheses after the `in` operator if there's more operations, as it has really high precedence.
Something like `for(var/obj/structure/cable/C in (T.contents - src))`

Reply:
something going bad, i cant profile any thing.

Reply:
i tested 
```
/obj/thing/verb/verb_list_test(n as num)
	set src in world
	while(n)
		n--
		list_test_1()
		list_test_2()
	world<<world.realtime

/obj/thing/proc/list_test_1()
	set src in world
	for(var/obj/O in loc.contents)
		if(O == src) continue
		bit++

/obj/thing/proc/list_test_2()
	set src in world
	for(var/obj/O in loc.contents - src)
		bit++
```
and get 
![image](https://user-images.githubusercontent.com/7734424/78075286-ee6e5a00-73ac-11ea-91b2-267dabecb123.png)
in pile of 10 items

seems like with no check is better.